jerry-core/ecma/builtin-objects/ecma-builtin-number.c

```diff
@@ -91,10 +92,140 @@ ecma_builtin_number_dispatch_construct (const ecma_value_t *arguments_list_p, /*
   }
 } /* ecma_builtin_number_dispatch_construct */
 
+#if ENABLED (JERRY_ES2015_BUILTIN)
+/**
+ * The Number object 'isFinite' routine
+ *
+ * See also:
+ *          ECMA-262 v6, 20.1.2.2
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_number_object_is_finite (ecma_value_t this_arg, /**< this argument */
+                                      ecma_value_t arg) /**< routine's argument */
+{
+  JERRY_UNUSED (this_arg);
+
+  if (ecma_is_value_number (arg))
+  {
+    ecma_number_t num = ecma_get_number_from_value (arg);
+    if (!(ecma_number_is_nan (num) || ecma_number_is_infinity (num)))
+    {
+      return ECMA_VALUE_TRUE;
+    }
+  }
+  return ECMA_VALUE_FALSE;
+} /* ecma_builtin_number_object_is_finite */
+
+/**
+ * The Number object 'isNan' routine
+ *
+ * See also:
+ *          ECMA-262 v6, 20.1.2.4
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_number_object_is_nan (ecma_value_t this_arg, /**< this argument */
+                                   ecma_value_t arg) /**< routine's argument */
+{
+  JERRY_UNUSED (this_arg);
+
+  if (ecma_is_value_number (arg))
+  {
+    ecma_number_t num_val = ecma_get_number_from_value (arg);
+
+    if (ecma_number_is_nan (num_val))
+    {
+      return ECMA_VALUE_TRUE;
+    }
+  }
+  return ECMA_VALUE_FALSE;
+} /* ecma_builtin_number_object_is_nan */
+
+/**
+ * The Number object 'isInteger' routine
+ *
+ * See also:
+ *          ECMA-262 v6, 20.1.2.3
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_number_object_is_integer (ecma_value_t this_arg, /**< this argument */
+                                       ecma_value_t arg) /**< routine's argument */
+{
+  JERRY_UNUSED (this_arg);
+
+  if (ecma_is_value_number (arg))
+  {
+    ecma_number_t num = ecma_get_number_from_value (arg);
+
+    if (ecma_number_is_nan (num) || ecma_number_is_infinity (num))
+    {
+      return ECMA_VALUE_FALSE;
+    }
+    return ECMA_VALUE_TRUE;
+  }
+  ecma_number_t integ = floor (fabs (arg));
```

Comment:
This seems a bit awkward: at this point the `arg` is not a number value, but it is used as a number.
AFAIK this calculation is only valid if the `arg` is a number and thus it should be in the `if` block above.
Also the calculation of `ToInteger` is incorrect, this will always be a positive number. The sign should be kept. Fortunately the `ecma_number_trunc` can be used for the same operation which does this correctly.